### PR TITLE
Feature/grey vest 2

### PR DESCRIPTION
**Pull request by michsa**
description

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:exclamation: This PR is BIG (+387 -360)  
Please keep it below 500 net changes</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: d7e1e22de93a937221e13fa9a3c9ebfba8e2d8f0


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0e335ba75159feea02e45da8ac53df14b0925754
</p>

File: src/themes/base/CheckboxList.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import F from 'futil'
+import _ from 'lodash/fp'
+import { ColumnList, Flex } from 'grey-vest'
+
+let CheckboxList = ({ options, value, onChange, ...props }) => (
+  <ColumnList gap={1} {...props}>
```
Comment (daedalus28) [inline src/themes/base/CheckboxList.js]: all of these gaps need updating

File: src/utils/react.js [hunk 2]
```diff
@@ -18,3 +18,13 @@ export let expandProp = _.flow(
   (key, fn) => F.expandObjectBy(key, F.whenExists(fn)),
   mapProps
 )
+
+export let defaultProps = props => BaseComponent => {
```
Comment (daedalus28) [inline src/utils/react.js:22]: why didn't we just use [`defaultProps` from `recompose`](https://github.com/acdlite/recompose/blob/master/docs/API.md#defaultprops)?

File: src/exampleTypes/ResultTable/ResultTableFooter.js [hunk 3]
```diff
@@ -12,12 +11,7 @@ let ResultTableFooter = ({ tree, node, pageSizeOptions }) => {
       page={node.page || 1}
       onChangePage={page => tree.mutate(node.path, { page })}
       pageSize={node.pageSize}
-      onChangePageSize={pageSize => {
-        tree.mutate(node.path, {
-          pageSize,
-          page: _.ceil((getFromContext('startRecord') || 0) / pageSize) || 1,
```
[the remark this answers is not in the document]
Comment (daedalus28) [reply src/exampleTypes/ResultTable/ResultTableFooter.js:18]: Seems to still be working correctly when I pull this locally

**Comment by daedalus28** [comment]
I'm not sure how, but there are a ton of conflicts from this to the v3 branch...

**Pull request merged**